### PR TITLE
add websockets query endpoint

### PR DESCRIPTION
**Pull request by xvrl**

### CODE REVIEW

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java [hunk 1]
```diff
@@ -229,6 +234,52 @@ public void configureBaseApplication(Configurable<?> config, Map<String, String>
...
+                @SuppressWarnings("unchecked")
+                public <T> T getEndpointInstance(Class<T> endpointClass)
+                    throws InstantiationException {
+                  return (T) new WSQueryEndpoint(
+                      getJsonMapper(),
+                      new StatementParser(ksqlEngine),
+                      ksqlEngine,
+                      MoreExecutors.listeningDecorator(
+                          Executors.newScheduledThreadPool(
+                              config.getInt(KsqlRestConfig.KSQL_WEBSOCKETS_NUM_THREADS),
```
[the remark this answers is not in the document]
Comment (xvrl) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java]: no, this is a global 5-thread pool to handle all the connections.

[the remark this answers is not in the document]
Comment (xvrl) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java]: oops, you're right need to move this out of the callback

[the remark this answers is not in the document]
Comment (apurvam) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java]: nice catch @rodesai !

**Comment by dguy** [comment]
testing?

**Pull request merged**